proto/tendermint/abci/types.proto

```diff
@@ -230,12 +230,6 @@ message ResponseCheckTx {
   repeated Event events     = 7
       [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
   string codespace = 8;
-  string sender    = 9;
-  int64  priority  = 10;
-
-  // mempool_error is set by CometBFT.
-  // ABCI applictions creating a ResponseCheckTX should not set mempool_error.
-  string mempool_error = 11;
```

Comment:
I know `v0.38.x` is ABCI-breaking w.r.t. `v0.37.x`, so this change should not be problematic.
However, we should add somewhere in the UPGRADING notes that we are removing fields from `ResponseCheckTx`.
Could you please add a line in #33 to not forget to do it? Alternatively, open an issue and hook it up to #33. Thanks

Reply:
The protobuf documentation indicates reserving field numbers as a possible solution here

https://protobuf.dev/programming-guides/proto/#reserved

Reply:
Yeah, we've used that in ABCI. I think it is a good idea.
So, can we... ?
* reserve those numbers here, maybe with a comment explaining why
* add a note in #33 to also mention these fields are disappearing in `v0.38.x`

Reply:
Please ignore bullet 2 above. I see you already did it in your latest commit

Reply:
Thanks for pointing this out @sergio-mena! I already updated the UPGRADING notes.

Reply:
That's a good point, @lasarojc! Afaiu, this could have been a security issue in proto2, but not in proto3, where every field is implicitly optional. So the new generated code will ignore data with the old fields that may come from an application. We may still want to mark these fields as reserved so that nobody tries to use them in an incompatible way. What do you think?

Reply:
Thanks for looking into it. I think it is a good idea to ensure that the field ids don't get reused and avoid any clashes.

Reply:
So at the end I reverted back the changes and marked these fields as reserved in the proto file. Thanks @lasarojc and @sergio-mena!